ui/apps/platform/src/Containers/NetworkGraph/flows/AdvancedFlowsFilter.tsx

```diff
@@ -0,0 +1,149 @@
+    setFilters,
+}: AdvancedFlowsFilterProps): React.ReactElement {
+    // derived state
+    const selections = filtersToSelections(filters);
+
+    // component state
+    const [isFilterDropdownOpen, setIsFilterDropdownOpen] = React.useState(false);
+    const {
+        isOpen: isPortsSelectOpen,
+        onToggle: onTogglePortsSelect,
+        onSelect: onSelectPorts,
+    } = useMultiSelect(handlePortsSelect, filters.ports, false);
+
+    // setters
+    const onFilterDropdownToggle = (isOpen: boolean) => {
+        setIsFilterDropdownOpen(isOpen);
+    };
+    const onTrafficFilterSelect = (event, selection) => {
+        if (selections.includes(selection)) {
+            setFilters((prevFilters) => {
+                const prevSelection = filtersToSelections(prevFilters);
+                const newSelection = prevSelection.filter((item) => item !== selection);
+                const newFilters = selectionsToFilters(newSelection);
+                return newFilters;
+            });
+        } else {
+            setFilters((prevFilters) => {
+                const prevSelection = filtersToSelections(prevFilters);
+                const newSelection = [...prevSelection, selection] as string[];
+                const newFilters = selectionsToFilters(newSelection);
+                return newFilters;
+            });
+        }
+    };
+    function handlePortsSelect(selection) {
+        setFilters((prevFilters) => {
+            const newFilters = { ...prevFilters };
+            newFilters.ports = selection;
+            return newFilters;
+        });
+    }
+
+    return (
+        <Select
+            className="advanced-flows-filters-select"
+            variant={SelectVariant.checkbox}
+            onToggle={onFilterDropdownToggle}
+            onSelect={onTrafficFilterSelect}
+            selections={selections}
+            isOpen={isFilterDropdownOpen}
+            placeholderText="Advanced"
+            aria-labelledby="advanced-flows-filters-select"
+            isGrouped
+            position={SelectPosition.right}
+        >
+            <SelectGroup label="Deployment flows">
+                <SelectOption value="anomalous">Anomalous flows</SelectOption>
+                <SelectOption value="baseline">Baselined flows</SelectOption>
+            </SelectGroup>
+            <SelectGroup label="Flow directionality">
+                <SelectOption value="ingress">Ingress (inbound)</SelectOption>
+                <SelectOption value="egress">Egress (outbound)</SelectOption>
+            </SelectGroup>
+            <SelectGroup label="Protocols">
+                <SelectOption value="TCP">TCP</SelectOption>
+                <SelectOption value="UDP">UDP</SelectOption>
+            </SelectGroup>
+            <SelectGroup label="Ports">
+                <Select
+                    className="pf-u-px-md"
+                    variant={SelectVariant.typeaheadMulti}
+                    aria-label="Select one or more days"
```

Comment:
The copy-paste gremlin is messing with the aircraft engine again.

```suggestion
                    aria-label="Select one or more ports"
```